### ISSUE TITLE
compute_rbf_centers builds an n_centers^n_features Cartesian grid, not n_features*n_centers as documented

### ISSUE THREAD

**Issue by fundthmcalculus** [fundthmcalculus/tribble-fis#130]
## Repro

Surfaced while sweeping \`TribbleRegressor\`'s construction hyperparameters against N-CMAPSS RUL data (grad-school repo, \`FuzzySystemsExperiments/cmapss_rul.py\`, PR #110).

\`\`\`python
model = TribbleRegressor(
    tsk_order="1st",  # or "full-2nd" -- anything but "0th"
    n_gaussians=5, top_p=0.9,
    consequent_basis="gaussian-rbf",
)
model.fit(X_train, y_train)  # X_train: 26320 rows, 15 selected top_features_
\`\`\`

\`\`\`
numpy._core._exceptions._ArrayMemoryError: Unable to allocate 41.2 TiB
for an array with shape (26320, 14348907, 15) and data type float64
\`\`\`

## Root cause

\`compute_rbf_centers\` (\`regression.py:853\`) builds centers via a full Cartesian-product grid across every feature dimension:

\`\`\`python
def compute_rbf_centers(X: ndarray, n_centers: int = 5) -> ndarray:
    """...
    Returns:
        (n_centers_total, n_features) array of RBF centers.
    """
    ...
    centers.append(feat_quantiles)
    # Create a grid of all combinations (Cartesian product across features)
    return np.column_stack(np.meshgrid(*centers, indexing='ij')).reshape(-1, n_features)
\`\`\`

\`TribbleRegressor\`'s own docstring for \`rbf_n_centers\` says this "produces n_features * n_centers total centers" -- but \`np.meshgrid\` over all \`n_features\` axes produces \`n_centers ** n_features\` points, not \`n_features * n_centers\`. With the default \`rbf_n_centers=3\` and just 15 selected features (unremarkable for this regressor -- \`top_features_\` routinely lands in the 10-20 range), that's \`3**15 = 14,348,907\` centers, each requiring a distance computation against every training row and every feature in \`_gaussian_rbf_features\` -- the resulting \`(n_samples, n_centers_total, n_features)\` intermediate array is what OOMs.

This isn't a \`tsk_order\`-specific issue -- it reproduces at \`tsk_order="1st"\` too. It only avoided the blowup at \`tsk_order="0th"\` in my testing, apparently because the 0th-order code path in \`solve_tsk_consequents\`/\`build_consequent_features\` never actually calls into \`compute_rbf_centers\`/\`build_consequent_features\` with the RBF basis for that order (a 0th-order TSK has no consequent features to build in the first place, so the buggy path is simply never reached -- not because centers are computed correctly at that order).

## Suggested fix

Docstring vs. implementation disagree on the intended scaling. If linear-in-features growth is the actual design intent (\`rbf_n_centers\` centers *per feature*, not a joint grid), the fix is to *not* take a Cartesian product -- e.g. stack each feature's own quantile centers independently (shape \`(n_features, n_centers)\` used per-feature in \`_gaussian_rbf_features\`, or place centers via k-means/quantiles in the joint space capped at a fixed total count regardless of dimensionality) rather than \`np.meshgrid\` across every axis. Either way, \`compute_rbf_centers\` needs a hard cap or a non-exponential construction so it can't be handed an ordinary-sized \`top_features_\` list and explode.

## Scope note

Not blocking for PR #110 -- I've excluded \`consequent_basis="gaussian-rbf"\` from that PR's construction hyperparameter sweep entirely rather than work around it there.

**Comment by fundthmcalculus** [comment]
Confirmed fixed — pulled latest \`main\` (058501f) into the grad-school DOE and re-tested the exact config that used to OOM:

\`\`\`python
TribbleRegressor(tsk_order="full-2nd", n_gaussians=5, consequent_basis="gaussian-rbf", ...)
\`\`\`

Now completes in ~0.65s instead of requesting 41TB. Swept \`rbf_gamma\`/\`rbf_n_centers\` across all 3 pipelines from the DOE while I was at it: best tuned gaussian-rbf result lands close to but consistently just behind the raw/orthogonal-basis champion (e.g. 12.48 vs. 8.83 on the widest pipeline; 19.35 vs. 19.12 and 21.91 vs. 21.66 on the other two, at `rbf_gamma≈0.003-0.01`, `rbf_n_centers=8`). So the basis is safe to use now, just not a winner on this dataset -- not reporting that as a further bug, just closing the loop.
